File: Ex1/ex1a.c
#include <stdio.h>

int main() {
    // Arithmetic Operations
    int num1, num2;
    printf("Enter two integers: ");
    scanf("%d %d", &num1, &num2);

    int sum = num1 + num2;
    int difference = num1 - num2;
    int product = num1 * num2;
    double quotient = 0.0;
    if (num2 != 0) {
        quotient = (double)num1 / num2;
    }

    printf("Arithmetic Operations:\n");
    printf("Sum: %d\n", sum);
    printf("Difference: %d\n", difference);
    printf("Product: %d\n", product);
    if (num2 != 0)
        printf("Quotient: %.2f\n", quotient);
    else
        printf("Quotient: undefined (division by zero)\n");

    // Relational and Logical Operations
    int a, b, c;
    printf("\nEnter three integers: ");
    scanf("%d %d %d", &a, &b, &c);

    // Relational operators
    int max = (a > b && a > c) ? a : (b > c) ? b : c;
    int min = (a < b && a < c) ? a : (b < c) ? b : c;

    // Logical operators
    int isEven = (num1 % 2 == 0) && (num2 % 2 == 0);
    int isPositive = (a > 0) && (b > 0) && (c > 0);

    printf("\nRelational and Logical Operations:\n");
    printf("Maximum among %d, %d, %d: %d\n", a, b, c, max);
    printf("Minimum among %d, %d, %d: %d\n", a, b, c, min);

    printf("Both numbers are%s even.\n", isEven ? "" : " not");
    printf("All numbers are%s positive.\n", isPositive ? "" : " not");

    return 0;
}

File: Ex1/ex1a.py
def main():
    num1, num2 = map(int, input("Enter two integers: ").split())

    total = num1 + num2
    difference = num1 - num2
    product = num1 * num2
    quotient = num1 / num2

    print("Arithmetic Operations:")
    print(f"Sum: {total}")
    print(f"Difference: {difference}")
    print(f"Product: {product}")
    print(f"Quotient: {quotient:.2f}")

    a, b, c = map(int, input("\nEnter three integers: ").split())
    maximum = max(a, b, c)
    minimum = min(a, b, c)
    is_even = num1 % 2 == 0 and num2 % 2 == 0
    is_positive = a > 0 and b > 0 and c > 0

    print("\nRelational and Logical Operations:")
    print(f"Maximum among {a}, {b}, {c}: {maximum}")
    print(f"Minimum among {a}, {b}, {c}: {minimum}")
    print(f"Both numbers are{' ' if is_even else ' not '}even.")
    print(f"All numbers are{' ' if is_positive else ' not '}positive.")

if __name__ == "__main__":
    main()
